h/views/activity.py

```diff
@@ -256,13 +256,22 @@ def domain(user):
                 return None
             return urlparse.urlparse(user.uri).netloc
 
-        annotation_count = None
+        public_annotation_count = None
+        if self.request.feature('total_shared_annotations'):
+            public_annotation_count = self.request.find_service(name='user').public_annotation_count(self.user.userid)
+
+        private_annotation_count = None
+        group_annotation_count = None
         if self.request.feature('total_shared_annotations'):
-            annotation_count = self.request.find_service(name='user').annotation_count(self.user.userid)
+            if self.request.authenticated_userid == self.user.userid:
+                private_annotation_count = self.request.find_service(name='user').private_annotation_count(self.user.userid)
+                group_annotation_count = self.request.find_service(name='user').all_groups_annotation_count(self.user.userid) - public_annotation_count
 
         result['user'] = {
             'name': self.user.display_name or self.user.username,
-            'annotation_count': annotation_count,
+            'public_annotation_count': public_annotation_count,
```

Comment:
It would be good to wrap the stats here up into a single `stats` object that gets passed into the template, rather than making this view method more and more complicated for every new stat we pass in. This would simplify the tests, as well.